File: samples/test/cross_lang/insertion_sort/InsertionSort.php
<?php 
	function insertionSort($arr) {
		$i = 1;
		while ($i < count($arr)) {
			$j = 0;
			while ($j < $i && $arr[$j] < $arr[$i]) {
				$j++;
			}
			$temp = $arr[$i];
			for ($k = $i; $k > $j; $k-- ) {
				$arr[$k] = $arr[$k-1];
			}
			$arr[$j] = $temp;
			$i++;
		}
	}
?>

File: samples/test/cross_lang/insertion_sort/InsertionSort.java
public class InsertionSort {
	void sort(int[] array) {
		int i, j, temp, k;
		i = 1;
		while (i < array.length) {
			j = 0;
			while (j < i && array[j] < array[i]) {
				j++;
			}
			temp = array[i];
			for (k = i; k > j; k-- ) {
				array[k] = array[k-1];
			}
			array[j] = temp;
			i++;
		}
	}
}